lib/Transforms/IPO/StructFieldCacheAnalysis.cpp

```diff
@@ -17,280 +17,31 @@
-  void analyzeUsersOfStructPointerValue(const Value *V);
-
-  /// Obtain which field the instruction is accessing and return no val if not
-  /// accessing any struct field
-  Optional<FieldNumType> getAccessFieldNum(const Instruction *I) const;
-
-  /// Obtain total number of instructions that access the struct fields
-  // unsigned getTotalNumFieldAccess() const { return
-  // LoadStoreFieldAccessMap.size() + CallInstFieldAccessMap.size(); }
-  unsigned getTotalNumFieldAccess() const {
-    return LoadStoreFieldAccessMap.size();
-  }
-
-  /// Obtain execution count for the BasicBlock/Instruction from profiling info,
-  /// if any
-  /// %{
-  Optional<uint64_t> getExecutionCount(const BasicBlock *BB) const {
-    return StructManager->getExecutionCount(BB);
-  }
-  Optional<uint64_t> getExecutionCount(const Instruction *I) const {
-    return StructManager->getExecutionCount(I->getParent());
-  }
-  /// %}
-
-  /// Print all instructions that access any struct field
-  void debugPrintAllStructAccesses(raw_ostream &OS);
-
-  /// For stats
-  /// %{
-  void addStats(unsigned Category, unsigned Opcode = 0) {
-    StatCounts[Category]++;
-    if (Category == StructFieldAccessManager::DebugStats::DS_GepUnknownUse) {
-      if (UnknownOpcodes.find(Opcode) == UnknownOpcodes.end())
-        UnknownOpcodes[Opcode] = 0;
-      else
-        UnknownOpcodes[Opcode]++;
-    }
-  }
-  unsigned getStats(unsigned Category) const { return StatCounts[Category]; }
-  /// %}
-
-  void printUnknownOpcodes(raw_ostream &OS) const {
-    if (UnknownOpcodes.size() == 0)
-      return;
-    OS << "Unknown opcodes stats: \n";
-    for (auto &it : UnknownOpcodes) {
-      OS << "Opcode " << it.first << ": " << it.second << " times\n";
-    }
-  }
-
-private:
-  bool Eligiblity;
-  const Module &CurrentModule;
-  const StructType *StructureType;
-  const StructFieldAccessManager::StructDefinitionType StructDefinition;
-  const DICompositeType *DebugInfo;
-  unsigned NumElements;
-  const StructFieldAccessManager *StructManager;
-
-  /// A map records all load/store instructions accessing which field of the
-  /// structure
-  std::unordered_map<const Instruction *, unsigned> LoadStoreFieldAccessMap;
-  std::vector<unsigned> StatCounts;
-  std::unordered_map<unsigned, unsigned> UnknownOpcodes;
-
-private:
-  /// Calculate which field of the struct is the GEP pointing to, from
-  /// GetElementPtrInst or GEPOperator
-  FieldNumType calculateFieldNumFromGEP(const User *U) const;
-
-  /// Record all users of a GEP instruction/operator that calculates the address
-  /// of a field.
-  void addFieldAccessFromGEP(const User *U);
-
-  /// Record an access pattern in the data structure for a load/store
-  void addFieldAccessNum(const Instruction *I, FieldNumType FieldNum);
-};
-
 /// This class is inherited from AssemblyAnnotationWriter and used
-/// to print annotated information on IR
+/// to print annotated information on IR. This class is private to
```

Comment:
I should've remove this sentence because I have already move this class into a cpp file. Again, I've updated this change in the next pull request. Thanks!